user-service/repositories/user_repository.go

```diff
@@ -0,0 +1,108 @@
+package repositories
+
+import (
+	"context"
+	"errors"
+
+	"github.com/vanjmali/spotlite/user-service/entities"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+var (
+	TokenExpiredErr = errors.New("invalid token")
+)
+
+type UserRepository struct {
+	DbName   string
+	CollName string
+	Client   *mongo.Client
+}
+
+func NewRepository(dbName string, collName string, c *mongo.Client) *UserRepository {
+	r := UserRepository{Client: c, DbName: dbName, CollName: collName}
+	return &r
+}
+
+// Create func, inserts a new user into the database,
+func (r *UserRepository) Create(ctx context.Context, user entities.User) error {
+	c := r.Client.Database(r.DbName).Collection(r.CollName)
+
+	_, err := c.InsertOne(ctx, user)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ActiveAndRevokeToken func, that activates the account and revokes the token in one database trip,
+func (r *UserRepository) ActiveAndRevokeToken(ctx context.Context, token string) error {
+	c := r.Client.Database(r.DbName).Collection(r.CollName)
+
+	// define filtering parameters,
+	filter := bson.M{
+		"account_status": "INACTIVE",
+		"token.type":     "ACCOUNT_VERIFICATION",
+		"token.content":  token,
+	}
+
+	// define set (set a field value to a new one) and unset (fully remove a field) operations,
+	update := bson.M{
+		"$set":   bson.M{"account_status": "ACTIVE"},
+		"$unset": bson.M{"token": ""},
+	}
```

Comment:
Filtering should use constants instead of hardcoded, making it easier to keep track of them.

```suggestion
	filter := bson.M{
		"account_status": entities.StatusInactive,
		"token.type":     entities.AccountVerification,
		"token.content":  token,
	}
```